src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -107,6 +110,15 @@ public bool IsProcessingEnumerable()
             return (JsonTypeInfo.PropertyInfoForTypeInfo.ConverterStrategy & ConverterStrategy.Enumerable) != 0;
         }
 
+        public void MarkRequiredPropertyAsRead(JsonPropertyInfo propertyInfo)
+        {
+            if (propertyInfo.IsRequired)
```

Comment:
Perhaps move this check outside of the method call and replace this with an assert? In 99% cases it would be a no-op, so best avoid the method call altogether.

Reply:
I'll mark it with AggressiveInlining which will have the same effect